### PR TITLE
Replace Reachability framework with one that correctly determines network status #99

### PR DESCRIPTION
**Pull request by einsteinx2**
I added a hello endpoint on our new backend that just returns the word "hello". When the reachability class determines the network is available, we poll that address immediately to see if we get the correct response. If it succeeds, we send the network available notification to begin syncing. If it fails, we retry every second until it either succeeds or the reachability class determines the network is unavailable.

This should fix the issue whereby the network interface is ready, but network connectivity isn't, causing initial syncs to fail.

Also I moved the Reachability.swift file into our project instead of using it as a Carthage framework since it's literally 1 file.

### CODE REVIEW

File: Balance/Shared/Data Model/Singleton/NetworkStatus.swift [hunk 1]
```diff
@@ -34,13 +33,60 @@ class NetworkStatus {
...
     fileprivate func becameUnreachable(reachability: Reachability) {
+        cancelCheckHost()
         NotificationCenter.postOnMainThread(name: Notifications.NetworkBecameUnreachable)
     }
+    
+    @objc fileprivate func checkHost() {
+        cancelCheckHost()
+        
+        async {
+            let url = URL(string: "https://balance-server.appspot.com/hello")!
```
Comment (wolffan) [inline Balance/Shared/Data Model/Singleton/NetworkStatus.swift]: this has a problem which is it relies on our site.

Comment (wolffan) [reply Balance/Shared/Data Model/Singleton/NetworkStatus.swift]: I guess we need to add the server check in another place then this will be valid

**Comment by wolffan** [comment]
My only request will be to move tests into the project as well (if the project had tests)

File: Balance/Shared/Data Model/Singleton/NetworkStatus.swift (hunk 1 above)
Comment (einsteinx2) [reply Balance/Shared/Data Model/Singleton/NetworkStatus.swift]: It relies on our backend server. If that goes down, the app will have problems all over like not being able to sign in to your balance profile. Hypothetically if the backend is down and you're already signed in, and your coinbase token is still valid, then we could sync all your accounts anyway, but I thought it was a reasonable URL to check. We should be shooting for almost 100% availability there anyway.

And what do you mean about adding the server check in another place?

**Comment by einsteinx2** [comment]
I didn't write any tests ;)

Let's talk tomorrow. I didn't even think of this as something that would be particularly testable...

**Comment by wolffan** [comment]
Updated with the test file
Could we merge the ethplore PR so we can take advantage of the changes that include the test setup? ATM there's no test target (so tests won't run)

**Comment by einsteinx2** [comment]
I merged the ethplorer request. Please approve this one and I'll merge it tomorrow.

**Comment by wolffan** [comment]
@einsteinx2 didn't finish to tweak the circle thing yesterday (i'll have the master branch ready for when you're awake)

**Pull request merged**